### PR TITLE
<code object>::replace()

### PR DESCRIPTION
**Pull request by dannasman**
Hello there!

I started working on issue #4022  and was just wondering whether this is the right place to implement the replace() function.

### CODE REVIEW

**Comment by youknowone** [comment]
Yes, exactly!

**Comment by dannasman** [comment]
Should I wrap the function parameters with `OptionalArg`? There is quite a lot of optional parameters.  Also what should the parameter types be. For example if you want to replace `flags` field of the code object should the parameter passed to `replace()` be `OptionalArg<u16>` or `OptionalArg<CodeFlags>`?

**Comment by youknowone** [comment]
You'd better to check `#[derive(FromArgs)]`. `OptionalArg` in function signature only supports optional positional args, but you need optional keyword args.
I prefer `CodeFlags` to `u16` when possible. RustPython automatically can make additional check during its conversion.

**Comment by dannasman** [comment]
I managed to implement the method so that it passes `test_replace` in `test_code.py` for all 10 attributes that have a `#[pygetset]` function in `code.rs` (rest I commented out for the duration of testing).

File: vm/src/builtins/code.rs [hunk 1]
```diff
@@ -5,15 +5,39 @@
...
-    function::FuncArgs,
+    function::{FuncArgs, OptionalArg},
     AsObject, Context, Py, PyObject, PyObjectRef, PyPayload, PyRef, PyResult, VirtualMachine,
 };
 use num_traits::Zero;
 use std::{borrow::Borrow, fmt, ops::Deref};
 
+#[derive(FromArgs)]
+pub struct ReplaceArgs {
+    #[pyarg(any, optional)]
```
Comment (youknowone) [inline vm/src/builtins/code.rs:19]:
```suggestion
    #[pyarg(named, optional)]
```
for every lines. because it doesn't take positional argument

**Review by youknowone — CHANGES_REQUESTED** [review]
other parts looks good, thank you!

**Review by youknowone — APPROVED** [review]
Great! Thank you for contributing!

**Comment by dannasman** [comment]
Thank you for the guidance!

**Pull request merged**